glean-core/android/src/main/java/mozilla/telemetry/glean/private/StringMetricType.kt

```diff
@@ -23,17 +23,23 @@ import mozilla.telemetry.glean.rust.toBoolean
  *
  * The string API only exposes the [set] method, which takes care of validating the input
  * data and making sure that limits are enforced.
+ *
+ * The internal constructor is only used by labeled metrics directly.
  */
-class StringMetricType(
-    disabled: Boolean,
-    category: String,
-    lifetime: Lifetime,
-    name: String,
+class StringMetricType internal constructor(
+    var handle: Long,
     val sendInPings: List<String>
 ) {
-    private var handle: Long
-
-    init {
+    /**
+     * The public constructor used by automatically generated metrics.
+     */
+    constructor(
+        disabled: Boolean,
+        category: String,
+        lifetime: Lifetime,
+        name: String,
+        sendInPings: List<String>
+    ) : this(0, sendInPings) {
```

Comment:
nit: since we've been bitten by this before in glean-ac, can you use named arguments here, i.e. `this(handle = 0, sendInPings = sendInPings)`?